modules/lib/core/index.d.ts

```diff
@@ -140,6 +135,27 @@ export interface Content<
     variantOf?: string;
 }
 
+export interface FragmentContent extends CommonContent {
+    fragment: LayoutComponent | PartComponent
+    type: 'portal:fragment'
+}
+
+export interface PageContent<
+    Data = Record<string, unknown>,
+    Type extends string = string,
+    Page extends Component = Component,
+> extends CommonContent {
+    data: Data;
+    page: Page;
+    type: Type;
+}
+
+export type Content<
```

Comment:
This will be a problem.

If I say I want an article-content by doing `contentLib.get<Content<Article>>()`, this will not give an `Content<Article>` anymore. It will give you the type `FragmentContent | Content<Article>`.

It will insist to make me **always** have to check `Content.type` if something is `FragmentContent` even if I told it the result is `Content<Article>`.

I will repost my solution from my last review here for you to consider again:

```diff
export interface Content<
    Data = Record<string, unknown>,
    Type extends string = string,
    Page extends Component = Component,
> {
    _id: string;
    _name: string;
    _path: string;
    _score?: number;
    creator: UserKey;
    modifier?: UserKey;
    createdTime: string;
    modifiedTime?: string;
    owner: string;
    data: Data;
    type: Type;
    displayName: string;
    hasChildren: boolean;
    language?: string;
    valid: boolean;
    originProject?: string;
    childOrder?: string;
    _sort?: object[];
    page: Page;
    x: XpXData;
    attachments: Record<string, Attachment>;
    publish?: PublishInfo;
    workflow?: {
        state: 'IN_PROGRESS' | 'PENDING_APPROVAL' | 'REJECTED' | 'READY';
        checks?: Record<string, 'PENDING' | 'REJECTED' | 'APPROVED'>;
    };
    inherit?: ('CONTENT' | 'PARENT' | 'NAME' | 'SORT')[];
    variantOf?: string;
+   fragment: Type extends "portal:fragment" 
+      ? LayoutComponent | PartComponent
+      : undefined;
}
```

Reply:
What about this madness? Not certain I like it...
<img width="677" alt="Screenshot 2023-05-02 at 15 04 13" src="https://user-images.githubusercontent.com/118847/235674390-f8401764-95c5-494a-93ba-e64038ad2668.png">